pkg/ingester/stream.go

```diff
@@ -150,11 +157,25 @@ func (s *stream) Push(_ context.Context, entries []logproto.Entry) error {
 		}()
 	}
 
-	if appendErr == chunkenc.ErrOutOfOrder {
-		return httpgrpc.Errorf(http.StatusBadRequest, "entry out of order for stream: %s", client.FromLabelAdaptersToLabels(s.labels).String())
+	if len(failedEntriesWithError) > 0 {
+		// return bad http status request response with all failed entries
+		buf := bytes.Buffer{}
+		streamName := client.FromLabelAdaptersToLabels(s.labels).String()
+
+		for _, entryWithError := range failedEntriesWithError {
+			if entryWithError.e == chunkenc.ErrOutOfOrder {
+				fmt.Fprintf(&buf,
+					"entry with timestamp %s ignored, reason: '%s' for stream: %s,\n",
+					entryWithError.entry.Timestamp.String(), entryWithError.e, streamName)
+			}
+		}
+
+		fmt.Fprintf(&buf, "total ignored: %d out of %d", len(failedEntriesWithError), len(entries))
+
+		return httpgrpc.Errorf(http.StatusBadRequest, buf.String())
```

Comment:
My concern here is what happens when there were no out of order errors. Before, that would get returned as an internal error to the user, but now any failure here gets passed back as a 4xx Bad Request. 

I'm not really sure of the best way to handle this but maybe if _none_ of the errors were an `ErrOutOfOrder` then we can just return the last error received rather than wrapping it in a Bad Request error.  

Maybe this is being too nitpicky? What do you think?

Reply:
hmm. good point @rfratto . What happens when one of the other error codes, like ErrInvalidSize is hit? Shouldn't this be a bad request as well? Or is ErrOutOfOrder the only user error possible?  

Anyway, what makes this error handling a bit funny, is that in theory you can have multiple reasons why an append fails, but the original function only find the type of the last error important. It's pretending to be atomic based on the function signature.

Reply:
From what I can tell from the code, `ErrOutOfOrder` is the only user error possible from doing an append. 

And yeah, the old design of ignoring all the errors is interesting. We could have a composite error type that collects all the errors, but I'm not sure it's that important. If the error gets reported to the user as a 5xx, they probably don't need to know how many errors were actually generated.

Reply:
i've added another commit to make failed appends due to the other possible error codes visible in the 4XX msg.